core/src/main/java/org/apache/gravitino/tag/TagManager.java

```diff
@@ -226,6 +227,42 @@ public MetadataObject[] listMetadataObjectsForTag(String metalake, String name)
         });
   }
 
+  public MetadataObject[] listMetadataObjectsForTags(String metalake, String[] names)
+      throws NoSuchTagException {
+    if (names == null || names.length == 0) {
+      return new MetadataObject[0];
+    }
+    NameIdentifier[] tagIds =
+        Arrays.stream(names)
+            .map(name -> NameIdentifierUtil.ofTag(metalake, name))
+            .toArray(NameIdentifier[]::new);
+    NameIdentifier lockId = tagIds[0];
+    return TreeLockUtils.doWithTreeLock(
+        lockId,
+        LockType.READ,
```

Comment:
Locking only one tag seems unable to solve the concurrency issues of other tags, do we have a better way to acquire locks for batch operations? @yuqi1129